dsmiddleware/splitcall/splitcall.go

```diff
@@ -0,0 +1,118 @@
+package splitcall
+
+import (
+	"context"
+
+	"go.mercari.io/datastore"
+)
+
+var _ datastore.Middleware = &splitHandler{}
+
+// New split call middleware will be returns.
+func New(opts ...Option) datastore.Middleware {
+	sh := &splitHandler{splitThreshold: 1000}
+	for _, opt := range opts {
+		opt.Apply(sh)
+	}
+	if sh.logf == nil {
+		sh.logf = func(ctx context.Context, format string, args ...interface{}) {}
+	}
+
+	return sh
+}
+
+// A Option is an option for splitcall.
+type Option interface {
+	Apply(*splitHandler)
+}
+
+type splitHandler struct {
+	splitThreshold int
+
+	logf func(ctx context.Context, format string, args ...interface{})
+}
+
+func (sh *splitHandler) AllocateIDs(info *datastore.MiddlewareInfo, keys []datastore.Key) ([]datastore.Key, error) {
+	return info.Next.AllocateIDs(info, keys)
+}
+
+func (sh *splitHandler) PutMultiWithoutTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) ([]datastore.Key, error) {
+	return info.Next.PutMultiWithoutTx(info, keys, psList)
+}
+
+func (sh *splitHandler) PutMultiWithTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) ([]datastore.PendingKey, error) {
+	return info.Next.PutMultiWithTx(info, keys, psList)
+}
+
+func (sh *splitHandler) GetMultiWithoutTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) error {
+	sh.logf(info.Context, "process %d keys", len(keys))
```

Comment:
Is this log output permanently?

Reply:
yes. default implementation is `func(ctx context.Context, format string, args ...interface{}) {}`